sdk/python/kfp/components/types/type_utils.py

```diff
@@ -283,10 +316,40 @@ def verify_type_compatibility(
     return types_are_compatible
 
 
-def check_parameter_type_compatibility(
+def check_artifact_type_compatibility(given_type: str,
+                                      given_is_artifact_list: bool,
+                                      expected_type: str,
+                                      expected_is_artifact_list: bool) -> bool:
+    given_schema_title, given_schema_version = given_type.split('@')
+    expected_schema_title, expected_schema_version = expected_type.split('@')
+    same_list_of_artifacts_status = expected_is_artifact_list == given_is_artifact_list
+    if not same_list_of_artifacts_status:
+        return False
+    elif artifact_types.Artifact.schema_title in {
+            given_schema_title, expected_schema_title
+    }:
+        return True
+    else:
+        schema_title_compatible = given_schema_title == expected_schema_title
+        schema_version_compatible = given_schema_version.split(
+            '.')[0] == expected_schema_version.split('.')[0]
+
+        return schema_title_compatible and schema_version_compatible
+
+
+def check_parameter_type_compatibility(given_type: str,
+                                       expected_type: str) -> bool:
+    if isinstance(given_type, str) and isinstance(expected_type, str):
+        return given_type == expected_type
+    else:
+        return check_v1_struct_parameter_type_compatibility(
+            given_type, expected_type)
+
+
+def check_v1_struct_parameter_type_compatibility(
```

Comment:
Improved variable names, function names, and code modularity throughout this Python module to promote clarity as the type checks become higher dimensional. Adding `is_artifact_list` checks adds an additional dimension.